cpp/src/backends/protocol/otf_message.cc

```diff
@@ -1,32 +1,45 @@
-#include <arpa/inet.h>
-#include <sys/socket.h>
-#include <glog/logging.h>
-
 #include "src/backends/protocol/otf_message.hh"
 
+const std::string CONTENT_TYPE_SUFFIX = ":contentType";
+
 namespace torchserve {
-  byte_buffer OTFMessage::CreateLoadModelResponse(StatusCode code, const std::string& message) {
-    LoadModelResponse response = {
-      code, 
-      static_cast<int>(message.length()), 
-      message
-    };
-    std::byte msg[sizeof(LoadModelResponse)];
-    std::memcpy(msg, &response, sizeof(LoadModelResponse));
-    byte_buffer response_byte_buffer;
-    std::copy(response_byte_buffer.begin(), response_byte_buffer.end(), msg);
-    return response_byte_buffer;
+  void OTFMessage::EncodeLoadModelResponse(std::unique_ptr<torchserve::LoadModelResponse> response, char* data) {
+    char* p = data;
+    // Serialize response code
+    int32_t s_code = htonl(response->code);
+    memcpy(p, &s_code, sizeof(s_code));
+    p += sizeof(s_code);
+    // Serialize response message length
+    int32_t resp_length = htonl(response->length);
+    memcpy(p, &resp_length, sizeof(resp_length));
+    p += sizeof(resp_length);
+    // Serialize response message
+    strcpy(p, response->buf.c_str());
+    p += response->length;
+    // Expectation from frontend deserializer is a -1
+    // at the end of a LoadModelResponse
+    int32_t no_predict = htonl(response->predictions);
```

Comment:
The flag of response ending should not be defined in LoadModelResponse.

Reply:
this change is from #1814. @maaquib

Reply:
I am not sure what the alternative would be. If this is not defined we would have to change the frontend

Reply:
Fixed in #1814